Adding blog on custom detection rules in Security Analytics (#2425)
opensearch-project/project-website: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md

---
layout: post
title:  "Building custom threat detection rules with OpenSearch Security Analytics"
authors:
- xeniatup
- jimishs
- sandeshkr419
date: 2023-11-28
categories:
 - feature
 - technical-posts

> meta_keywords: threat detection, security analytics, SIEM, detection rules

Reviewer: Please update the following: 

**meta_keywords:** OpenSearch security analytics, threat detection, security analytics plugin, detection rules

4. In the **Details** section, select a log type, rule severity, and rule status. 

![Rule details section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/details.png)

5. Complete **Selection_1** in the **Detection** section to define the criteria used for the rule.

You can add multiple **Selections** with multiple Maps for the log fields. **Map** is a directory that contains key-value pairs. **Modifiers** transform values and lists or convert them into regular expressions. To learn more about the Sigma rule format, go to [documentation](https://github.com/SigmaHQ/sigma). 

- Create a Map with **Key**: `UserID`, **Modifier**: `contains`, and **Value**: `<your organization domain name>`.
- Create a Map with **Key**: `Recipient`, **Modifier**: `endswith`, and **List**: `<values that include the domains to be detected>`. 

![Selection_1 section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/selection1.png)

6. Create another **Selection** where **Key**: `UserId` is a list of emails belonging to the marketing team by uploading the list from a .csv (or .txt) file. 

> 

Reviewer 2: Delete extra blank line

![Upload a file dialog](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/upload-a-file.png)

After uploading the list, **Selection_2** will look like this:

![Selection_2 section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/selection2.png)

> 7. Enter  `Selection_1 AND NOT include Selection_2` in the **Condition** fields to define how each **Selection** will contribute to the final query . 

Reviewer 2: ```suggestion
7. Enter  `Selection_1 AND NOT include Selection_2` in the **Condition** fields to define how each **Selection** will contribute to the final query. 
```

> The UI allows the combining of multiple selections using operators, including `AND`, `AND NOT`, `OR`, and `OR NOT`. 

Reviewer 2: ```suggestion
The user interface (UI) allows for the combining of multiple selections using operators, including `AND`, `AND NOT`, `OR`, and `OR NOT`. 
```

> **Pro tip**: If you’re more comfortable using YAML formats, you can switch to the YAML editor on the top of the page under the page title at any point and write a **Condition** as code without losing the context.

Reviewer 2: ```suggestion
**Pro tip**: If you’re more comfortable using YAML formats, you can switch to the YAML editor at the top of the page under the page title at any point and write a **Condition** as code without losing the context.
```

![Condition section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/condition.png) 

Once a custom rule is added, it will show up on the **Detection rules** page and can be used in any new or existing detector for customized fine-grained threat detection.

### What are the benefits?

> A custom approach to detection rules that are tailored to your unique business cases helps target specific malicious activity across a range of cloud environments and troubleshoot application-specific threats. OpenSearch Security Analytics provides the capability to configure alerts and customize notification messages, helping users to find threats that need further investigation. By using the custom detection rules workflow, you can expand the detection criteria to include custom application log sources and monitor them for specific threat patterns. You can also use correlation rules to find patterns between standard and custom application logs. 

Reviewer 2: ```suggestion
A custom approach to detection rules that are tailored to your unique business cases helps target specific malicious activity across a range of cloud environments and troubleshoot application-specific threats. OpenSearch Security Analytics provides the capability to configure alerts and customize notification messages, helping you to find threats that need further investigation. By using the custom detection rules workflow, you can expand the detection criteria to include custom application log sources and monitor them for specific threat patterns. You can also use correlation rules to find patterns between standard and custom application logs. 
```

> Integrating custom rules to the threat detection strategies empowers your security teams to enhance coverage for detecting security threats across a variety of log sources. Custom rules can be written with a deep understanding of an organization's infrastructure, making them more precise in identifying specific threats and reducing false positives. Combining custom and pre-packaged rules for threat detection strikes a balance between tailored and standardized protection.

Reviewer 2: ```suggestion
Integrating custom rules into the threat detection strategies empowers your security teams to enhance coverage for detecting security threats across a variety of log sources. Custom rules can be written with a deep understanding of an organization's infrastructure, making them more precise in identifying specific threats and reducing false positives. Combining custom and prepackaged rules for threat detection strikes a balance between tailored and standardized protection.
```

> To get started on creating a custom detection rule for a custom log source or application log, refer to [documentation](https://opensearch.org/docs/latest/security-analytics/usage/rules/#creating-detection-rules). To learn more about security analytics and its capabilities, see the [documentation](https://opensearch.org/docs/latest/security-analytics/index/).

Reviewer 2: ```suggestion
To get started on creating a custom detection rule for a custom log source or application log, refer to [documentation](https://opensearch.org/docs/latest/security-analytics/usage/rules/#creating-detection-rules). To learn more about security analytics and its capabilities, see the [Working with detection rules documentation](https://opensearch.org/docs/latest/security-analytics/index/).
```
